jupyterhub_config.py

```diff
@@ -46,6 +46,13 @@
         'profile_options': {
             'image': {
                 'display_name': 'Image',
+                'free_form': {
```

Comment:
@yuvipanda @batpad are you okay naming this `other_choice`?

I think the "other choice" name provides intuition that:
- its one additional choice to `choices`
- of a special "free input" kind
- that its just one single choice addition

I don't think naming this `free_form` would provide as much intuition on what the configuration is about compared to `other_choice`. And if not, it would making this config harder to understand intuitively, which could make it harder to and setup and update over time.

Reply:
> @yuvipanda @batpad are you okay naming this other_choice?

`other_choice` sounds great to me :-)

Will wait a bit to see if @yuvipanda has any strong opinions, but I can go ahead and make that change.

Do the other variable names inside the object sound okay? Maybe be nice to make all naming changes in one go.

Thanks again for the thoughtfulness around this.

Reply:
- `enabled` :+1:
- `display_name` :+1: - consistent with other naming
- `match_regex` :thinking: --- works, but I can also see value in having `validation_` be involved in this naming, if it is `validation_match_regex` or `validation_regex`
- `validation_message` :+1: 
- `kubespawner_override` :+1: - consistent with other naming

Reply:
Nice!

@consideRatio I have made a commit renaming `free_form` to `other_choice` and `match_regex` to `validation_match_regex`.

> match_regex 🤔 --- works, but I can also see value in having validation_ be involved in this naming, if it is validation_match_regex or validation_regex. I think I favor validation_match_regex.

I definitely like prefixing with `validation` since we also have `validation_message` and also STRONGLY agree that `validation` should not be further nested. Personally, I think I'd prefer the slightly less verbose `validation_regex` to `validation_match_regex` as `validation_regex` sounds pretty unambiguous to me. But also not a hard opinion, and I'm not as familiar with general naming conventions here, so went with your suggestion here. If you do agree / feel that `validation_regex` is fine, am happy to change that.

Thank you again.

Reply:
Ah, forgot to mention - also changed the `name` for the input field that's submitted with the POST request from `profile-option-{{ profile.slug }}-{{ k }}-other` to `profile-option-{{ profile.slug }}-{{ k }}-other-choice` to keep that consistent with the name of the option.

@consideRatio would be great to get your 👀 and 👍 / 👎 on the naming of this as well, since that's the interface between the front and backend and it would be nice to be thoughtful about what we call this.

Reply:
> Personally, I think I'd prefer the slightly less verbose `validation_regex` to `validation_match_regex` as `validation_regex` sounds pretty unambiguous to me.

I think you are right that it sounds pretty unambiguous and I couldn't figure out how to misunderstand it!

> If you do agree / feel that `validation_regex` is fine, am happy to change that.

Thank you for thinking about naming things so thoroughly!!! Let's go for it!

Reply:
> Thank you for thinking about naming things so thoroughly @batpad!!! Let's go for it!

Made this change :-)